app/build.gradle

```diff
@@ -12,6 +12,22 @@ buildscript {
 apply plugin: 'com.android.application'
 apply plugin: 'com.google.firebase.firebase-perf'
 apply plugin: 'io.fabric'
+apply plugin: 'checkstyle'
+
+task checkstyle(type: Checkstyle) {
+    configFile file("${project.rootDir}/ci/checkstyle.xml")
+    source 'src'
+    include '**/*.java'
+    exclude '**/gen/**'
+    exclude '**/test/**'
+    exclude '**/R.java'
+    exclude '**/BuildConfig.java'
+
+    classpath = files()
+}
+
+// If checkstyle fails we can't generate a build
+preBuild.dependsOn('checkstyle')
```

Comment:
Doing this we don't need to add this custom gradle task to our Continuous Integration System, right ?

Reply:
Exactly, checkstyle always runs before a build